src/clj/game/core/engine.clj

```diff
@@ -439,9 +439,12 @@
   ([state side card message choices ability args]
    (let [f #(resolve-ability state side ability card [%])]
      (show-prompt state side (:eid ability) card message choices f
-                  (if-let [f (:cancel-effect args)]
-                    (assoc args :cancel-effect #(f state side (:eid ability) card [%]))
-                    args)))))
+                  (let [c (:cancel-effect args)]
+                    (assoc args :cancel-effect 
+                             (if c (if (fn? c)
+                                       #(do (c state side (:eid ability) card [%]) (effect-completed state side (:eid ability)))
```

Comment:
Don't pass in the `eid` if the function won't complete it. Use `(make-eid state (:eid ability))` instead.